packages/@aws-cdk/aws-cloudwatch/test/test.metrics.ts

```diff
@@ -124,4 +142,82 @@ export = {
 
     test.done();
   },
+
+  'throws error when there are more than 10 dimensions in dimensionsMap'(test: Test) {
+    test.throws(() => {
+      new Metric({
+        namespace: 'Test',
+        metricName: 'ACount',
+        period: cdk.Duration.minutes(10),
+        dimensionsMap: {
+          dimensionA: 'value1',
+          dimensionB: 'value2',
+          dimensionC: 'value3',
+          dimensionD: 'value4',
+          dimensionE: 'value5',
+          dimensionF: 'value6',
+          dimensionG: 'value7',
+          dimensionH: 'value8',
+          dimensionI: 'value9',
+          dimensionJ: 'value10',
+          dimensionK: 'value11',
+        },
+      } );
+    }, /The maximum number of dimensions is 10, received 11/);
+
+    test.done();
+  },
+
+  'can make metric with dimensionsMap property'(test: Test) {
+    const m = new Metric({
+      namespace: 'Test',
+      metricName: 'Metric',
+      period: cdk.Duration.minutes(10),
+      dimensionsMap: {
+        dimensionA: 'value1',
+        dimensionB: 'value2',
+      },
+    });
+
+    expect(m).toMatch({
+      namespace: 'Test',
+      metricName: 'Metric',
+      period: {
+        amount: 10,
+        unit: {
+          label: 'minutes',
+          isoLabel: 'M',
+          inMillis: 60000,
+        },
+      },
+      dimensions: {
+        dimensionA: 'value1',
+        dimensionB: 'value2',
+      },
+      statistic: 'Average',
+    });
+
+    test.done();
+  },
+
+  '"with" with a different dimensions property'(test: Test) {
+    const dims = {
+      dimensionA: 'value1',
+    };
+
+    const m = new Metric({
+      namespace: 'Test',
+      metricName: 'Metric',
+      period: cdk.Duration.minutes(10),
+      dimensionsMap: dims,
+    });
+
+    test.notEqual(m.with({
+      dimensions: {
+        dimensionB: 'value2',
+      },
+    }).dimensions, dims);
```

Comment:
Can we change this to a positive assertion instead?

```suggestion
    test.deepEqual(metric.with({
      dimensionsMap: {
        dimensionB: 'value2',
      },
    }).dimensions, {
      dimensionB: 'value2',
    });
```

(My mind immediately wants to add a test case "What if `with()` overrides an existing dimension key", but this change is not really about that, so I'll let it slide 😜)